packages/core/src/components/button/_common.scss

```diff
@@ -393,6 +393,7 @@ $button-intents: (
 
     &:disabled,
     &.pt-disabled {
+      background: none;
```

Comment:
In the `datetime` package, we typically handle this by adding `:not(.pt-disabled)` in all of the hover and active styles. I prefer what you have here though.